### PR TITLE
SPUI-512/513 > Portfolio with balance

### PR DESCRIPTION
**Pull request by VovaMedvediev**

### CODE REVIEW

File: lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart [hunk 1]
```diff
@@ -39,82 +39,130 @@ class PortfolioWithBalanceBody extends HookWidget {
...
+                Padding(
+                  padding: EdgeInsets.only(
+                    left: 24.w,
+                  ),
+                  child: Row(
+                    children: [
+                      Text(
+                        hidden.state ? '' : periodChange,
+                        style:
+                            sSubtitle3Style.copyWith(color: periodChangeColor),
```
Comment (eli1stark) [inline lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart:82]: comma

File: lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart [hunk 2]
```diff
@@ -39,82 +39,130 @@ class PortfolioWithBalanceBody extends HookWidget {
...
+                      ),
+                      const SpaceW10(),
+                      Text(
+                        'Today',
+                        style: sBodyText2Style.copyWith(color: colors.grey3),
```
Comment (eli1stark) [inline lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart:87]: comma

File: lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart [hunk 3]
```diff
@@ -39,82 +39,130 @@ class PortfolioWithBalanceBody extends HookWidget {
     final showZeroBalanceWallets = useProvider(showZeroBalanceWalletsStpod);
     final baseCurrency = useProvider(baseCurrencyPod);
     final clientDetail = useProvider(clientDetailPod);
+    final periodChange = _periodChange(
+      chart,
+      baseCurrency,
```
Comment (eli1stark) [inline lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart:44]: can be placed in one line for clarty

File: lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart [hunk 4]
```diff
@@ -39,82 +39,130 @@ class PortfolioWithBalanceBody extends HookWidget {
...
-              ),
+          ),
+          BalanceChart(
+            onCandleSelected: (ChartInfoModel? chartInfo) {
+              chartN.updateSelectedCandle(chartInfo?.right);
+            },
+            walletCreationDate: clientDetail.walletCreationDate,
+          ),
+          const SpaceH76(),
+          Padding(
```
Comment (eli1stark) [inline lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart:102]: repeated component #1

File: lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart [hunk 5]
```diff
@@ -39,82 +39,130 @@ class PortfolioWithBalanceBody extends HookWidget {
...
+                        ? 'HIDDEN'
+                        : _price(
+                            chart,
+                            itemsWithBalance,
+                            baseCurrency,
+                          ),
+                    style: sTextH1Style,
+                  ),
+                ),
+                Padding(
```
Comment (eli1stark) [inline lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart:73]: repeated component #1

File: lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart [hunk 6]
```diff
@@ -39,82 +39,130 @@ class PortfolioWithBalanceBody extends HookWidget {
...
-              ),
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          SizedBox(
+            height: 68.h,
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: [
+                Padding(
```
Comment (eli1stark) [inline lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart:58]: repeated component #1

File: lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart [hunk 7]
```diff
@@ -39,82 +39,130 @@ class PortfolioWithBalanceBody extends HookWidget {
...
+                secondaryText: '${item.assetBalance} ${item.symbol}',
+                onTap: () => _navigateToWallet(context, item),
+                color: colors.black,
+                removeDivider: item == itemsWithoutBalance.last,
               ),
-          ],
-        ),
+          if (!zeroBalanceWalletsEmpty(itemsWithoutBalance))
+            Padding(
+              padding: EdgeInsets.symmetric(vertical: 27.5.h),
```
Comment (eli1stark) [inline lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart:150]: comma

File: lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_header.dart [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    final colors = useProvider(sColorPod);
+
+    return SizedBox(
+      height: 120.h,
+      child: Column(
+        children: [
+          SizedBox(
```
Comment (eli1stark) [inline lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_header.dart:18]: use SpaceH64

**Review by eli1stark — CHANGES_REQUESTED** [review]

File: lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart (hunk 1 above)
Comment (VovaMedvediev) [reply lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart:82]: added

File: lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart (hunk 2 above)
Comment (VovaMedvediev) [reply lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart:87]: added

File: lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart (hunk 3 above)
Comment (VovaMedvediev) [reply lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart:44]: done

File: lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart (hunk 7 above)
Comment (VovaMedvediev) [reply lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart:150]: added

File: lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_header.dart (hunk 8 above)
Comment (VovaMedvediev) [reply lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_header.dart:18]: done

File: lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart (hunk 4 above)
Comment (VovaMedvediev) [reply lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart]: refactored

File: lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart (hunk 5 above)
Comment (VovaMedvediev) [reply lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart]: refactored

File: lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart (hunk 6 above)
Comment (VovaMedvediev) [reply lib/app/screens/portfolio/view/components/portfolio_with_balance/portfolio_with_balance_body.dart]: refactored

**Pull request merged**